File: VB-Examples/10_Conversion/ToPDF/SetImageQuality.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace SetImageQuality
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object
			Dim document As New Document()

			' Load a Word document from the specified file path
			document.LoadFromFile("..\..\..\..\..\..\..\Data\Template_Doc_1.doc", FileFormat.Doc)

			' Set the JPEG quality for the document to 40
			document.JPEGQuality = 40

			' Specify the file name for the resulting PDF
			Dim result As String = "Result-DocToPDFImageQuality.pdf"

			' Save the document to a PDF file with the specified image quality
			document.SaveToFile(result, FileFormat.PDF)

			' Dispose of the Document object to release resources
			document.Dispose()

			'Launch the Pdf file.
			WordDocViewer(result)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS-Examples/10_Conversion/SetImageQuality.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace SetImageQuality
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of the Document class
			Document document = new Document();

			// Load a Word document from the specified file path using the 'Doc' format
			document.LoadFromFile(@"..\..\..\..\..\..\..\Data\Template_Doc_1.doc", FileFormat.Doc);

			// Set the JPEG quality for saving images in the document to 40%
			document.JPEGQuality = 40;

			// Specify the output file name for the converted PDF
			string result = "Result-DocToPDFImageQuality.pdf";

			// Save the document as a PDF using the specified file format
			document.SaveToFile(result, FileFormat.PDF);

			// Dispose of the document object to free up resources
			document.Dispose();

            //Launch the Pdf file.
            WordDocViewer(result);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
